app/controllers/users/users_controller.rb

```diff
@@ -0,0 +1,18 @@
+class UsersController < ApplicationController
+
+  def destroy
+    @user = User.find(params[:id])
+
+    @user.orders.destroy_all
```

Comment:
This shouldn't be needed from here if you have on orders on the user foreign key `on_delete: :cascade` & in the user model `has_many :orders, dependent: :delete_all`